src/waveform/waveformwidgetfactory.cpp

```diff
@@ -359,6 +364,16 @@ void WaveformWidgetFactory::addTimerListener(QWidget* pWidget) {
             Qt::DirectConnection);
 }
 
+
+void WaveformWidgetFactory::slotSkinLoaded() {
+    // This regenerates the waveforms twice because of a bug found on Windows
+    // where the first one fails.
+    // The problem is that the window of the widget thinks that it is not exposed.
+    // (https://doc.qt.io/qt-5/qwindow.html#exposeEvent )
+    setWidgetTypeFromConfig();
```

Comment:
Please guard this as well. 
Here in both cases the force flag is set in the later call. 
Is this intended?

Reply:
The force parameter was added simply so that it would not cause a no-op if the same waveform is already selected. It's just a "I don't know what's setup, but probably I need to force a replace".

Reply:
Ah, ok do it doesn't matter.
Can we always force than?